OpenAI-DotNet/Threads/MessageResponse.cs

```diff
@@ -93,13 +142,27 @@ public sealed class MessageResponse : BaseResponse
 
         public static implicit operator string(MessageResponse message) => message?.ToString();
 
+        public static implicit operator Message(MessageResponse response)
+            => response?.Content switch
+            {
+                string content => new(content, response.Role, response.Attachments, response.Metadata),
+                IReadOnlyList<Content> contents => new(contents, response.Role, response.Attachments, response.Metadata),
+                _ => null
+            };
+
         public override string ToString() => Id;
 
         /// <summary>
         /// Formats all of the <see cref="Content"/> items into a single string,
         /// putting each item on a new line.
         /// </summary>
         /// <returns><see cref="string"/> of all <see cref="Content"/>.</returns>
-        public string PrintContent() => string.Join("\n", Content.Select(content => content?.ToString()));
+        public string PrintContent()
+            => Content switch
+            {
+                string content => content,
+                IReadOnlyList<Content> contents => string.Join("\n", contents.Select(content => content?.ToString())),
```

Comment:
This does not seem to deserialise properly unfortunately. In the test `03_Threads.Test_04_02_ListMessages` on line 141-142, the message contents is printed to console but never asserted, so it seems the test passes but incorrectly. I added an assert  `Assert.IsNotEmpty(content);` which then causes the test to fail.

When debugging `PrintContent` i can see that when listing messages, the `Content` property is a JsonElement with a `ValueKind` = `Array`. The array then contains the `TextContent` structure that I think is expected (for a text message).

But, since the switch checks for an `IReadOnlyList<Content>` and not a `IReadOnlyList<TextContent>`, the condition defaults to the empty string.

I was not able to wrap my head around which contracts were used for sending and receiving data since they seem to differ. 

Anways, just thought I'd post the finding. You probably have a solution already :)

Reply:
I tested with JSON mode and didn't run into any trouble with my implementation in this PR

Reply:
I see. I really don't want to take up your time unnecessarily but just wanted to clarify my bloated post.

The line `Console.WriteLine($"[{threadMessage.Id}] {threadMessage.Role}: {threadMessage.PrintContent()}");` in the test is supposed to print the content of the threadMessage. The `PrintContent` value that is returned is never asserted to actually contain a value.

I expected this value to be something else than empty string. I expected the message strings previously created to be printed. But perhaps I am mistaken on what `PrintContent` is supposed to do.

Reply:
I'm aware. I haven't added the proper switching to print these to string yet. Well I have it locally but haven't pushed it up.